lora.cpp

```diff
@@ -227,4 +222,25 @@ void lora::sendUplink(lora::uplinkMessage msg)
 
   DEBUG_SERIAL_LOG.printf("Json Uplink: %s\r\n", output);
   sendUplink(output);
+}
+
+bool lora::isUplinkProcess()
+{
+  return uplinkProcess;
+}
+
+int lora::getRemainingDuration() {  
+  if (uplinkProcess) {
+    loraTimer -= millis() - lastHandledTime;
+    lastHandledTime = millis();
+
+    if (loraTimer <= 0)
+    {
+        uplinkProcess = false; 
+        DEBUG_SERIAL_LOG.println("Uplink routine complete");
+        return INT_MAX; 
+    }
+    return loraTimer;    
+  }
+  return INT_MAX; 
```

Comment:
Can you please explain how this function fits into things? I'm not sure that I understand what it's for. Thanks!